srf_generation/input_file_generation/realisation_to_srf.py

```diff
@@ -0,0 +1,292 @@
+        slip = (moment * 1.0e-20) / (aa * vs * vs * rho)
+
+    ###
+    ### file names
+    ###
+    srf_file = simulation_structure.get_srf_path(cs_root, name)
+    gsf_file = srf_file.replace(".srf", ".gsf")
+    makedirs(path.dirname(srf_file), exist_ok=True)
+
+    ###
+    ### create GSF
+    ###
+    with open(gsf_file, "w") as gsfp:
+        gsfp.write("# nstk= 1 ndip= 1\n")
+        gsfp.write(f"# flen= {dd:10.4f} fwid= {dd:10.4f}\n")
+        gsfp.write(
+            "# LON  LAT  DEP(km)  SUB_DX  SUB_DY  LOC_STK  LOC_DIP  LOC_RAKE  SLIP(cm)  INIT_TIME  SEG_NO\n"
+        )
+        gsfp.write("1\n")
+        gsfp.write(
+            f"{longitude:11.5f} {latitude:11.5f} {depth:8.4f} {dd:8.4f} {dd:8.4f} "
+            f"{strike:6.1f} {dip:6.1f} {rake:6.1f} {slip:8.2} {inittime:8.3f}    0\n"
+        )
+
+    ###
+    ### create SRF
+    ###
+    commands = [
+        binary_version.get_unversioned_bin(GENERICSLIP2SRF),
+        f"infile={gsf_file}",
+        f"outfile={srf_file}",
+        "outbin=0",
+        f"stype={stype}",
+        f"dt={dt}",
+        "plane_header=1",
+        f"risetime={risetime}",
+        "risetimefac=1.0",
+        "risetimedep=0.0",
+    ]
+    call(commands, stderr=PIPE)
+
+    ###
+    ### save STOCH
+    ###
+    stoch_file = simulation_structure.get_stoch_path(cs_root, name)
+    gen_stoch(stoch_file, srf_file)
+
+    ###
+    ### save INFO
+    ###
+    create_info_file(
+        srf_file,
+        1,
+        magnitude,
+        rake,
+        dt,
+        vs=vs,
+        rho=rho,
+        centroid_depth=depth,
+        lon=longitude,
+        lat=latitude,
+    )
+
+    # location of resulting SRF file
+    return srf_file
+
+
+def generate_sim_params_yaml(sim_params_file: str, parameters: Dict[str, Any]):
+    makedirs(path.dirname(sim_params_file), exist_ok=True)
+
+    sim_params = {}
+    for key, value in parameters.items():
+        if key in HF_RUN_PARAMS:
+            if "hf" not in sim_params.keys():
+                sim_params["hf"] = {}
+            sim_params["hf"][key] = value
+
+        elif key in BB_RUN_PARAMS:
+            if "bb" not in sim_params.keys():
+                sim_params["bb"] = {}
+            sim_params["bb"][key] = value
+
+        elif key in LF_RUN_PARAMS:
+            if "emod3d" not in sim_params.keys():
+                sim_params["emod3d"] = {}
+            sim_params["emod3d"][key] = value
+
+        else:
+            sim_params[key] = value
+
+    with open(sim_params_file, "w") as spf:
+        yaml.dump(sim_params, spf)
+
+
+def load_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("realisation_file", type=path.abspath)
+    parser.add_argument(
+        "-c", "--cybershake_root", type=path.abspath, default=path.abspath(".")
+    )
+    return parser.parse_args()
+
+
+def main():
+    args = load_args()
+    rel_df: pd.DataFrame = pd.read_csv(args.realisation_file)
+    realisation = rel_df.to_dict(orient="records")[0]
```

Comment:
I presume this is an intermediate script? Just does the first realisation?

Reply:
The csv only has one record in it, as you requested

Reply:
Oh, easy then.